pkgs/build-support/cc-wrapper/cc-wrapper.sh

```diff
@@ -81,29 +70,29 @@ fi
 
 # Optionally filter out paths not refering to the store.
 if [[ "${NIX_ENFORCE_PURITY:-}" = 1 && -n "$NIX_STORE" ]]; then
-    rest=()
+    kept=()
     nParams=${#params[@]}
     declare -i n=0
     while (( "$n" < "$nParams" )); do
         p=${params[n]}
         p2=${params[n+1]:-} # handle `p` being last one
-        if [ "${p:0:3}" = -L/ ] && badPath "${p:2}"; then
-            skip "${p:2}"
-        elif [ "$p" = -L ] && badPath "$p2"; then
-            n+=1; skip "$p2"
-        elif [ "${p:0:3}" = -I/ ] && badPath "${p:2}"; then
-            skip "${p:2}"
-        elif [ "$p" = -I ] && badPath "$p2"; then
-            n+=1; skip "$p2"
-        elif [ "$p" = -isystem ] && badPath "$p2"; then
-            n+=1; skip "$p2"
-        else
-            rest+=("$p")
-        fi
         n+=1
+
+        path=""
+        case "$p" in
+            -I/* | -L/*) path=${p:2} ;;
+            -I | -L | -isystem) path=$p2 n+=1 ;;
+        esac
+
+        if badPath "$path"; then
```

Comment:
```suggestion
        if test -N "$path" && badPath "$path"; then
```
I am not sure `badPath` should be happy with the empty path, an in any case I rather not rely on that.

Reply:
badPath is happy with the empty case since it returns early if $path != /* basically. But yeah I can be explicit about it.